src/main/java/matching_system/MatcherUITemplateJames.java

```diff
@@ -17,14 +17,19 @@ public class MatcherUITemplateJames extends JPanel implements ActionListener {
     private String requesterEmail;
     private Map<Integer, String> targetUserEmails = new HashMap<>();
 
+    /** A preset of UI panel that display 5 user info which are included in the matchResponseModel
+     * @param matcherUIController UIController to response to actions
+     * @param requesterEmail email of the requester
+     * @param matchUIPresenter the presenter that stores matchedRespondModel
+     * */
     public MatcherUITemplateJames(MatcherUIController matcherUIController,
                              String requesterEmail,
-                             MatchResponseModel matchResponseModel){
+                             MatchOutputBoundary matchUIPresenter){
```

Comment:
MatcherUITemplateJames cannot call MatchOutpurBoundary directly, but can call MatcherUIPresenter. So change the type of matchUIPresenter to MatchUIPresenter.

Reply:
Changed